llvm/test/CodeGen/AIE/aie2/schedule/interblock/liveness.mir

```diff
@@ -0,0 +1,151 @@
+# REQUIRES: asserts
+
+# These tests verify liveouts of every MachineBasicBlock in MachineFunction
+
+#     bb.0
+#     /  \
+# bb.1    bb.2
+#     \  /
+#     bb.3
+
+---
+name:            test_liveness_branch
+alignment:       16
+tracksRegLiveness: true
+body:             |
+  ; CHECK-LIVEOUTS-LABEL: test_liveness_branch:BB0
+  ; CHECK-LIVEOUTS-NEXT:   Live Registers: $r2 $r4 $r3 $lr $r5
+  ; CHECK-LIVEOUTS-NEXT: test_liveness_branch:BB1
+  ; CHECK-LIVEOUTS-NEXT:   Live Registers: $lr $r5 $r2
+  ; CHECK-LIVEOUTS-NEXT: test_liveness_branch:BB2
+  ; CHECK-LIVEOUTS-NEXT:   Live Registers: $lr $r5 $r2
+  ; CHECK-LIVEOUTS-NEXT: test_liveness_branch:BB3
+  ; CHECK-LIVEOUTS-NEXT:   Live Registers: (empty)
+
+  bb.0:
+    liveins: $r1, $r2, $r5
+    successors: %bb.1, %bb.2
+    $r0 = nsw ADD $r1, $r2, implicit-def $srcarry
+    $r3 = nsw ADD $r0, $r1, implicit-def $srcarry
+    $r4 = nsw ADC $r3, $r5, implicit-def $srcarry, implicit $srcarry
+    $r7 = MOV_mv_scl $srcarry
+    JNZ $r7, %bb.2
+    DelayedSchedBarrier
+  bb.1:
+    successors: %bb.3
+    liveins: $r2, $r4
+    $r5 = nsw ADD $r2, $r4, implicit-def $srcarry
+    J_jump_imm %bb.3
+    DelayedSchedBarrier
+  bb.2:
+    successors: %bb.3
+    liveins: $r3, $r4
+    $r1 = nsw ADD $r3, $r4, implicit-def $srcarry
+  bb.3:
+    liveins: $r2, $r5, $r6
+    $r1 = nsw ADD $r5, $r2, implicit-def $srcarry
+    RET implicit $lr
+    DelayedSchedBarrier implicit $r1 
+...
+
+#     bb.0
+#       |
+#     bb.1* (self loop)
+#       | 
+#     bb.2
+---
+name:            test_liveness_loop_1
+alignment:       16
+tracksRegLiveness: true
+body:             |
+  ; CHECK-LIVEOUTS-LABEL: test_liveness_loop_1:BB0
+  ; CHECK-LIVEOUTS-NEXT:   Live Registers: $r1 $amlh2 $amll2 $crfpmask $bml2 $bml1 $amll1 $amlh1 $bml0 $amll0 $amlh0 $lr
+  ; CHECK-LIVEOUTS-NEXT: test_liveness_loop_1:BB1
+  ; CHECK-LIVEOUTS-NEXT:   Live Registers: $r1 $amlh2 $amll2 $crfpmask $bml2 $bml1 $amll1 $amlh1 $bml0 $amll0 $amlh0 $lr
+  ; CHECK-LIVEOUTS-NEXT: test_liveness_loop_1:BB2
+  ; CHECK-LIVEOUTS-NEXT:   Live Registers: (empty)
+
+  bb.0:
+    liveins: $bml0, $bml1, $r0
+    successors: %bb.1
+    $bml2 = VADD_F $bml0, $bml1, $r0, implicit-def $srfpflags,  implicit $crfpmask
+    $r1 = MOV_mv_scl $r0
+  bb.1:
+    liveins: $bml0, $bml1, $bml2, $r1
+    successors: %bb.1, %bb.2
+    $bml3 = VADD_F $bml0, $bml2, $r1, implicit-def $srfpflags,  implicit $crfpmask
+    $bml1 = VADD_F $bml1, $bml3, $r1, implicit-def $srfpflags,  implicit $crfpmask
+    $bml2 = VADD_F $bml1, $bml2, $r1, implicit-def $srfpflags,  implicit $crfpmask
+    $r1 = ADD_add_r_ri $r1, -1, implicit-def $srcarry
+    JNZ $r1, %bb.1
+    DelayedSchedBarrier
+  bb.2:
+    liveins: $bml2
+    RET implicit $lr
+    DelayedSchedBarrier implicit $bml2
+...
+
+#     bb.0-----*
+#     /  \     |
+# bb.1    bb.2 |
+#     \  /     |
+#     bb.3-----*
+#       |
+#     bb.4
+---
+name:            test_liveness_loop_2
```

Comment:
It might give you a pass for the simple case immediately. Less investigation when the complicated test fails. smaller test files cause less merge conflicts. lit tests run in milliseconds